### PR TITLE
WIP: (unit-test-babel-passport): refactor model, add test config, con…

### PR DESCRIPTION
**Pull request by ezesundayeze**
This is a work in progress (WIP). I have added the following updates:

- Login,
- refactored the model
-  configured passport local strategy
- configured the test library (added tapejs)
- Adds test, but test isn't passing yet. Something more needs to happen.
-  other tiny chores
- I tried to research more about the package-lock.json being checked to version control and I think most people say it's good practice to have it here. The reason I added it. So, @Ani-Emmanuel , you were right. https://stackoverflow.com/questions/48524417/should-the-package-lock-json-file-be-added-to-gitignore

### CODE REVIEW

File: model/role.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const mongoose = require('mongoose')
+
```
Comment (Ani-Emmanuel) [inline model/role.js]: you need to import claim schema here because you referenced the schema

File: model/user.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const mongoose = require('mongoose')
+
```
Comment (Ani-Emmanuel) [inline model/user.js]: you need to import role schema here because you referenced the schema

Comment (Ani-Emmanuel) [reply model/user.js]: eg const Role = require("./role")

File: model/role.js (hunk 1 above)
Comment (Ani-Emmanuel) [reply model/role.js]: eg const Claim = require("./claim")

Comment (ezesundayeze) [reply model/role.js]: Oh, I see. 👍

Comment (ezesundayeze) [reply model/role.js]: Thanks

**Pull request merged**